src/test/java/frc/robot/subsystems/ColorSubsystemTest.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.subsystems;
+
+import org.junit.Test;
+
+import frc.robot.datasources.ColorSource;
+import com.revrobotics.ColorSensorV3.RawColor;
+import frc.robot.utils.ColorOptions;
+
+import org.junit.Before;
+import static org.junit.Assert.*;
+
+public class ColorSubsystemTest {
+
+  class MockColorSource implements ColorSource {
+
+    private RawColor rawColor;
+
+    MockColorSource(RawColor rawColor) {
+      this.rawColor = rawColor;
+    }
+
+    public RawColor getColor() {
+      return rawColor;
+    }
+  }
+
+  @Before
+  public void setup() {
+
+  }
+
+  @Test
+  public void test() {
+    ColorSubsystem cs = new ColorSubsystem(new MockColorSource(new RawColor(0, 0, 0, 0)));
+
+    ColorOptions co = cs.getColor();
+
+    assertEquals(ColorOptions.NONE, co);
+  }
+
+  @Test
+  public void testRed() {
+    RawColor rc = new RawColor(255, 153, 0, 0);
+    System.out.format("rc: Red:%d Green:%d Blue:%d\n", rc.red, rc.green, rc.blue);
+
+    ColorSubsystem cs = new ColorSubsystem(new MockColorSource(rc));
+
+    ColorOptions co = cs.getColor();
+    System.out.format("co: %s\n", co.toString());
+
+    assertEquals(ColorOptions.RED, co);
+  }
+
+  @Test
+  public void testGreen() {
+    RawColor rc = new RawColor(0, 255, 0, 0);
+    System.out.format("rc: Red:%d Green:%d Blue:%d\n", rc.red, rc.green, rc.blue);
+
+    ColorSubsystem cs = new ColorSubsystem(new MockColorSource(rc));
+
+    ColorOptions co = cs.getColor();
+    System.out.format("co: %s\n", co.toString());
+
+    assertEquals(ColorOptions.GREEN, co);
+  }
+
+  @Test
+  public void testCyan() {
+    RawColor rc = new RawColor(0, 255, 255, 0);
+    System.out.format("rc: Red:%d Green:%d Blue:%d\n", rc.red, rc.green, rc.blue);
+
+    ColorSubsystem cs = new ColorSubsystem(new MockColorSource(rc));
+
+    ColorOptions co = cs.getColor();
+    System.out.format("co: %s\n", co.toString());
+
+    assertEquals(ColorOptions.CYAN, co);
+  }
+
+  @Test
+  public void testYellow() {
+    RawColor rc = new RawColor(153, 255, 0, 0);
+    System.out.format("rc: Red:%d Green:%d Blue:%d\n", rc.red, rc.green, rc.blue);
+
+    ColorSubsystem cs = new ColorSubsystem(new MockColorSource(rc));
+
+    ColorOptions co = cs.getColor();
+    System.out.format("co: %s\n", co.toString());
+
+    assertEquals(ColorOptions.YELLOW, co);
+  }
+}
```

Comment:
In addition to the checks for the 4 intended colors, there need to be a set of tests to verify that other colors are correctly rejected.

Reply:
I picked 4 colors that are similar in RGB values to each of the 4 colors on the CP to confirm that the detection code is able to reject them.  It does.